include/dgl/runtime/tensordispatch.h

```diff
@@ -134,6 +134,66 @@ class TensorDispatcher {
     auto entry = entrypoints_[Op::kCUDACurrentStream];
     return FUNCCAST(tensoradapter::CUDACurrentStream, entry)();
   }
+
```

Comment:
> Maybe the easier way is to create a google doc to illustrate the diagram of the logic, 

In fact, the implementation logic of `CachingHostAllocator` is very detailed in the comment of pytorch src code. I think we don't need to create a doc to rephrase their design...

I tried to describe the main idea behind it briefly (maybe not brief enough 😞) and hope it can make future developers' life easier when dealing with this allocator.

Will slightly update again.